### PR TITLE
pylint: enable "missing-docstring" warning

### PR DESCRIPTION
**Pull request by zenhack**
There are a handful of other things in here that I encountered on the way; biproduct of combing over almost the entire codebase. I've tried to keep the commits coherent, so it should be relatively easy to tell what's happening where. The docstring additions are in their own commits with not much else.

### CODE REVIEW

File: tests/unit/class_resolver.py [hunk 1]
```diff
@@ -63,12 +73,14 @@ def test_class_resolver():
 
 
 def test_class_Obm():
+    """Test class_resovler with MockObm"""
```
Comment (naved001) [inline tests/unit/class_resolver.py]: class_reso**lv**er

File: tests/unit/class_resolver.py [hunk 2]
```diff
@@ -63,12 +73,14 @@ def test_class_resolver():
...
     build_class_map_for(model.Obm)
     assert concrete_class_for(model.Obm, mockapi_name + "obm/mock") \
         is hil.ext.obm.mock.MockObm
 
 
 def test_class_Switch():
+    """Test class_resovler with MockSwitch"""
```
Comment (naved001) [inline tests/unit/class_resolver.py]: class_reso**lv**er

File: tests/unit/deferred.py [hunk 3]
```diff
@@ -162,6 +196,7 @@ def new_nic(name):
 
 @pytest.fixture()
 def network():
+    """Create a test network (ans associated project) to work with."""
```
Comment (naved001) [inline tests/unit/deferred.py]: an**d**

File: hil/test_common.py [hunk 4]
```diff
@@ -208,16 +219,30 @@ def sample_obj(self):
...
 
     def test_repr(self):
+        """Test that printing the sample object doesn't explode.
+
+        Early on there were a few bugs where __repr__ threw an exception.
+        """
         print(self.sample_obj())
 
     def test_insert(self):
+        """Test inserting the sample object into the database.
```
Comment (zenhack) [inline hil/test_common.py]: The tests in tests/unit/model.py subclass this, and implement sample_obj.

File: tests/deployment/ipmi.py [hunk 5]
```diff
@@ -86,6 +86,9 @@ def test_node_set_bootdev(self):
                 api.node_set_bootdev(node.label, 'invalid-device')
 
         # register a node with erroneous ipmi details to raise OBMError
+        # XXX: In theory, this could actually be a real node; we should take
+        # some measure to ensure this never collides with something actually
```
Comment (zenhack) [inline tests/deployment/ipmi.py]: Yeah I figured we could do something like that; I just wanted to document the problem while I was in there.

Comment (zenhack) [inline tests/deployment/ipmi.py]: (We don't even have to haveit be random every time -- just do it once to make sure the user isn't going to cause a collision by accident).

File: tests/unit/api/main.py [hunk 6]
```diff
@@ -1193,6 +1386,7 @@ def test_headnode_detach_network_not_attached(self):
         api.headnode_detach_network('hn-0', 'hn-0-eth0')
 
     def test_headnode_detach_network_no_such_headnode(self):
+        """Detaching a non-existant headnode raises not found."""
```
Comment (zenhack) [inline tests/unit/api/main.py]: Yeah, apparently I just don't know how to spell this :/

**Comment by zenhack** [comment]
I think I've fixed everything. Re: splitting things up, will do that next time.

**Comment by naved001** [comment]
@SahilTikale @henn can we get a second review?

**Comment by naved001** [comment]
@jeremyfreudberg can you review this?

File: examples/dbinit.py [hunk 7]
```diff
@@ -27,6 +27,9 @@
 
 
 def hil(*args):
+    """Convienence function that calls the hil command line tool with
```
Comment (knikolla) [inline examples/dbinit.py]: nit: convenience

File: tests/unit/api/main.py [hunk 8]
```diff
@@ -1826,7 +2106,10 @@ def test_list_projects(self):
         ]
 
     def test_no_free_nodes(self):
-        assert json.loads(api.list_nodes("ree")) == []
```
Comment (knikolla) [inline tests/unit/api/main.py]: oh :/

**Comment by zenhack** [comment]
Quoting Kristi Nikolla (2017-09-01 15:24:19)

>    The refactorings sprinkled in made this a pain to review. Please keep
>    PRs to a single purpose.

Sorry -- I tried to keep things at least to their own commits, but when
you start writing the same docstring over and over it becomes apparent
that maybe you should just do the refactoring, and it's a little tricky
once you've already touched a bunch of other code. I will try to be more
judicious about this next time.

**Comment by zenhack** [comment]
Quoting Kristi Nikolla (2017-09-01 15:24:19)

>  def hil(*args):
> +    """Convienence function that calls the hil command line tool with
>
>    nit: convenience

Fixed.

**Comment by zenhack** [comment]
I'm going to click merge, since this has got two signoffs.

**Pull request merged**